internal/observability/tracing/tracing.go

```diff
@@ -83,14 +85,27 @@ func configureJaeger(ctx context.Context) error {
 }
 
 func configureOTLP(ctx context.Context) error {
-	conn, err := grpc.DialContext(ctx, conf.OTLP.CollectorEndpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
-	if err != nil {
-		return fmt.Errorf("failed to dial otlp collector: %w", err)
-	}
+	var exporter *otlptrace.Exporter
+	var err error
 
-	exporter, err := otlp.New(ctx, otlp.WithGRPCConn(conn))
-	if err != nil {
-		return fmt.Errorf("failed to create otlp exporter: %w", err)
+	switch conf.OTLP.Protocol {
+	case "grpc":
```

Comment:
Instead of using `SetDefaults`, how about handling the default protocol like this?

```suggestion
	case "", "grpc":
```

Reply:
we can use default in the switch structure for default grpc?

Reply:
I think it would be better to be explicit about the values that we support rather using `default` to mean "grpc" (that would mean that if you accidentally set it to `"http"` instead of `"http/protobuf"` you'd get gRPC rather than an error). It would probably be good to validate the protocol in the `(*Conf).Validate` method so that we fail sooner if the setting is invalid.